js/select2.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * @file
+ * Select2 integration.
+ */
+(function ($) {
+  'use strict';
+
+  Drupal.behaviors.select2 = {
+    attach: function (context) {
+      $.each(drupalSettings.select2, function(selector, settings) {
```

Comment:
I think its better to iterate over '[data-drupal-selector]' first and look for settings in drupalSettings later in conjunction with jquery once to avoid double initialization during ajax requets and/or big pipe.